metrics/metrics.go

```diff
@@ -139,14 +168,14 @@ func RegisterMetrics() {
 	prometheus.MustRegister(TotalCopProcHistogram)
 	prometheus.MustRegister(TotalCopWaitHistogram)
 	prometheus.MustRegister(HandleSchemaValidate)
-	prometheus.MustRegister(GRPCConnTransientFailureCounter)
 	prometheus.MustRegister(MaxProcs)
 	prometheus.MustRegister(GOGC)
 	prometheus.MustRegister(ConnIdleDurationHistogram)
 	prometheus.MustRegister(ServerInfo)
 	prometheus.MustRegister(TokenGauge)
 	prometheus.MustRegister(ConfigStatus)
 
-	tikvmetrics.InitMetrics("tidb", "tikvclient")
+	tikvmetrics.InitMetrics(TiDB, TiKVClient)
+	tikvmetrics.TiKVPanicCounter = initMetric(panicCounter.resetMetricDomain(TiDB, TiKVClient)) // reset tidb metrics for tikv metrics
```

Comment:
Why do we need to recreate the counterVec?

Reply:
Because `subsystem` param of the two Components(TiDB-Server,TiKV-Client-Go).